pilot/pkg/bootstrap/server.go

```diff
@@ -277,9 +301,12 @@ func (s *Server) initMesh(args *PilotArgs) error {
 	}
 
 	if mesh == nil {
+		var err error
 		// Config file either wasn't specified or failed to load - use a default mesh.
-		defaultMesh := model.DefaultMeshConfig()
-		mesh = &defaultMesh
+		if _, mesh, err = GetMeshConfig(s.kubeClient, kube.IstioNamespace, kube.IstioConfigMap); err != nil {
```

Comment:
When is the config file not specified (see below for current pilot deployment)? And when would that file read fail? 

https://github.com/istio/istio/blob/720be7a6b476f83229087c350b43c6556ccfd361/install/kubernetes/templates/istio-pilot.yaml.tmpl#L156-L158

https://github.com/istio/istio/blob/720be7a6b476f83229087c350b43c6556ccfd361/install/kubernetes/templates/istio-pilot.yaml.tmpl#L179-L181

Reply:
I think the configFile is used for VMs. In a K8s setup, it always fails, I tested it.

Reply:
Pilot discovery is not run for VMs, only proxy. 

You mean it can fail if the config map is not mounted ?

Reply:
I really don't see how this works - if the config map is not yet ready, but the mount and the load would fail.

I agree we shouldn't use the defaults silently - but fail if config map is missing, but I'm not sure this (reading using the API) solves much.

Reply:
`args.Mesh.ConfigFile` should default to `/etc/istio/config/mesh` per https://github.com/istio/istio/blob/720be7a6b476f83229087c350b43c6556ccfd361/pilot/cmd/pilot-discovery/main.go#L87. Is that path not available in pilot-discovery container?

Reply:
In fact the path was not available when running pilot from my mac. When running in a pod, the configMap gets mounted under /etc/istio/config, so there is no need to re-read the map.

The problem with the connectTimeout was in fact there are two of them in the config, one at the top level, and one in defaultConfig. Only the one at the top level is taken into account for CDS.

I can close this PR, unless we want to keep the ability of falling back on reading from a cluster config map when running pilot as a process locally. It seems to work fine.